sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -263,18 +299,31 @@ case class OptimizeSkewedJoin(conf: SQLConf) extends Rule[SparkPlan] {
     val shuffleStages = collectShuffleStages(plan)
 
     if (shuffleStages.length == 2) {
```

Comment:
why not we break this limitation first?

Reply:
Because this PR is not to address the case which has multiple SMJ. We have another PR to change this limitation:
1. `optimizeSingleStageSkewJoin`. This is the case one table is a bucket table and the SMJ is bucketing join with one side shuffle and skewing
2. `optimizeThreeShuffleStageSkewJoin`. This is to address three tables SMJ (Two SMJs in one stage and no one can be changed to BCJ in AQE).